examples/todo-app/web/src/context/TodoContext.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { TodoInterface } from "../interfaces/todo";
+import { LocalStorage, requestHandler } from "../utils";
+import { createTodoApi, getAllTodos } from "../api";
+import toast from "react-hot-toast";
+
+// Create a context to manage todos
+const TodoContext = createContext<{
+  todos: TodoInterface[];
+  loading: boolean;
+  createLoading: boolean;
+  changeTodo: (_todo: TodoInterface[]) => void;
+  createTodo: (title: string, description: string) => Promise<void>;
+}>({
+  todos: [],
+  loading: false,
+  createLoading: false,
+  createTodo: async () => {},
+  changeTodo: () => {},
+});
+
+// Create a hook to access the todos and related functions
+const useTodo = () => useContext(TodoContext);
+
+// Create a component that provides todos
+const TodoProvider: React.FC<{ children: React.ReactNode }> = ({
+  children,
+}) => {
+  const [todos, setTodos] = useState<TodoInterface[]>([]);
+  const [createLoading, setCreateloading] = useState<boolean>(false);
+  const [loading, setLoading] = useState<boolean>(false);
+
+  // function to create todo
+  const createTodo = async (title: string, description: string) => {
+    await requestHandler(
+      async () => await createTodoApi(title, description),
+      setCreateloading,
+      (res) => {
+        const data = res.data;
+        setTodos([...todos, data]);
```

Comment:
the newly created todo must come on the top. change `setTodos([...todos, data]);` to `setTodos([data, ...todos]);`